switch_mod/upgrade/upgrade_2_0_0b1.py

```diff
@@ -0,0 +1,274 @@
+"""
+Upgrade input directories from 2.0.0b0 to 2.0.0b1.
+Major changes are:
+* gen_tech files are merged into project_ files
+* The software version number is stored in an input file
+* modules.txt explicitly lists each module by its full name
+* lz_economic_multiplier is dropped from load_zones
+* proj_existing_builds is renamed to proj_existing_and_planned_builds
+
+API Synopsis:
+    import switch_mod.upgrade.upgrade_2_0_0b1 as upgrade_2_0_0b1
+    if upgrade_2_0_0b1.can_upgrade_inputs_dir(inputs_dir):
+        upgrade_2_0_0b1.upgrade_input_dir(inputs_dir):
+
+"""
+
+import os
+import shutil
+import pandas
+import argparse
+import switch_mod.upgrade
+
+upgrades_from = '2.0.0b0'
+upgrades_to = '2.0.0b1'
+
+rename_modules = {
+    'project.no_commit': 'operations.no_commit',
+    'project.unitcommit': 'operations.unitcommit',
+    'trans_build': 'investment.trans_build',
+    'trans_dispatch': 'operations.trans_dispatch',
+    'project.discrete_build': 'investment.proj_discrete_build',
+    'project.unitcommit.discrete': 'operations.unitcommit.discrete',
+    'fuel_cost': 'switch_mod.financials.fuel_flat_costs',
+    'fuel_markets': 'switch_mod.financials.fuel_markets',
+    'fuels': 'switch_mod.energy_sources'
+}
+module_prefix = 'switch_mod.'
+expand_modules = {
+    'switch_mod.operations.unitcommit': [
+        'switch_mod.operations.unitcommit.commit',
+        'switch_mod.operations.unitcommit.fuel_use'
+    ],
+    'switch_mod.project': [
+        'switch_mod.investment.proj_build',
+        'switch_mod.operations.proj_dispatch'
+    ]
+}
+core_modules = [
+    'switch_mod',
+    'switch_mod.timescales',
+    'switch_mod.financials',
+    'switch_mod.financials.minimize_cost',
+    'switch_mod.load_zones',
+    'switch_mod.energy_sources',
+    'switch_mod.investment.proj_build',
+    'switch_mod.operations.proj_dispatch',
+    'switch_mod.export'
+]
+
+def can_upgrade_inputs_dir(inputs_dir):
+    """
+    Determine if input directory can be upgraded with this script.
+    Returns True/False
+    """
+    version = switch_mod.upgrade.get_input_version(inputs_dir)
+    if version is None:
+        return False
+    return version == upgrades_from
+
+def upgrade_input_dir(inputs_dir, verbose=False):
+    """
+    Upgrade an input directory. If the directory has already 
+    been upgraded, this will have no effect.
+    """
+    if not can_upgrade_inputs_dir(inputs_dir):
+        if verbose:
+            print "Skipping upgrade for inputs directory {}.".format(inputs_dir)
+        return False
+
+    # Make a zip file backup before proceeding
+    switch_mod.upgrade.backup(inputs_dir)
+
+    # Does 'modules' need to get renamed to 'modules.txt'?
+    modules_path_old = os.path.join(inputs_dir, 'modules')
+    modules_path = os.path.join(inputs_dir, 'modules.txt')
+    if os.path.isfile(modules_path_old):
+        shutil.move(modules_path_old, modules_path)
+
+    ###
+    # Upgrade module listings
+    with open(modules_path, 'rb') as f:
```

Comment:
This should use the same logic and possibly the same arguments as solve.py to find the modules file (and maybe the inputs dir). The best way might be to share command line arguments with solve.py. This could be similar to how solve_scenarios.py calls solve.get_option_file_args().